src/openforms/js/components/admin/form_design/useConfirm.js

```diff
@@ -0,0 +1,57 @@
+import {useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import ActionButton from 'components/admin/forms/ActionButton';
+import {Modal} from 'components/admin/modals';
+
+const useConfirm = (message, title = '') => {
+  const [promise, setPromise] = useState(null);
+
+  const confirm = () =>
+    new Promise(resolve => {
+      setPromise({resolve});
+    });
+
+  const handleClose = () => {
+    setPromise(null);
+  };
+
+  const handleConfirm = () => {
+    promise?.resolve(true);
+    handleClose();
+  };
+
+  const handleCancel = () => {
+    promise?.resolve(false);
+    handleClose();
+  };
+
+  const ConfirmationModal = () => {
+    const intl = useIntl();
+    const confirmBtnText = intl.formatMessage({
+      description: 'Confirmation modal confirm button',
+      defaultMessage: 'Confirm',
+    });
+    const cancelBtnText = intl.formatMessage({
+      description: 'Confirmation modal cancel button',
+      defaultMessage: 'Cancel',
+    });
+    return (
+      <Modal
+        title={title}
+        isOpen={promise !== null}
+        contentModifiers={['with-form', 'small']}
+        closeModal={handleCancel}
+      >
+        <div style={{flexGrow: 1}}>{message}</div>
+        <div className="button-group">
```

Comment:
We could maybe reuse `submit-row`, to remove the need for extra styling for button-group

![image](https://github.com/user-attachments/assets/cee702f6-aaf0-4f98-abc0-0612fd6f17ef)

Reply:
Yeah i don't really like the styling of the `submit-row`. The weird extra border and background, extra padding and the uppercase button text.

I get your point, and maybe a modal action bar would be nice (in my head more like a bootstrap modal, where the padding between the `submit-row` and the modal would be removed). But then i would like to change that for all the modals.

For now i think the button group looks quite a bit better

Reply:
I've removed the `button-group` styling and added some modal specific styling. So this still doesn't use the `submit-row` class